src/esp/assets/BaseMesh.h

```diff
@@ -19,49 +25,141 @@
 namespace esp {
 namespace assets {
 
+/**
+  @brief Enumeration of mesh types supported by the simulator. Each entry
+  suggests a derived class of @ref BaseMesh implementing the specific storage
+  and processing interface to accomodate differing asset formats. Identifies the
+  derived variant of this object.
+  */
 enum SupportedMeshType {
+  /**
+   * @brief Undefined mesh types are created programmatically without a specific
+   * format or loaded from an unkown format. Support for this type and behavior
+   * is likely limited. Object type is likely @ref BaseMesh.
+   */
   NOT_DEFINED = -1,
+
+  /**
+   * @brief Instance meshes loaded from sources including segmented object
+   * identifier data (e.g. semantic data: chair, table, etc...). Sources include
+   * .ply files and reconstructions of Matterport scans. Object is likely of
+   * type @ref GenericInstanceMeshData or Mp3dInstanceMeshData.
+   */
   INSTANCE_MESH = 0,
+
+  /**
+   * @brief Meshes loaded from Replica dataset. Object is likely type @ref
+   * PTexMeshData.
+   */
   PTEX_MESH = 1,
+
+  /**
+   * @brief Meshes loaded from gltf format (i.e. .glb file). Object is likely
+   * type @ref GltfMeshData.
+   */
   GLTF_MESH = 2,
+
+  /**
+   * @brief Number of enumerated supported types.
+   */
   NUM_SUPPORTED_MESH_TYPES = 3,
 };
 
+/**
+ * @brief Base class for storing mesh asset data including geometry and
+ * topology. Also manages transfer of this data to GPU memory. Derived classes
+ * implement modifications to support specific mesh formats as enumerated by
+ * @ref SupportedMeshType.
+ */
 class BaseMesh {
  public:
+  /** @brief Constructor defining the @ref SupportedMeshType of this asset and
+   * likely identifying the derived type of this object.*/
   explicit BaseMesh(SupportedMeshType type) : type_(type){};
 
+  /** @brief Destructor */
   virtual ~BaseMesh(){};
 
+  /**
+   * @brief Set the @ref SupportedMeshType, @ref type_, of this object.
+   * @param type The desired @ref SupportedMeshType to set.
+   * @return Whether or not the set was successful.
+   */
   bool setMeshType(SupportedMeshType type);
+
+  /**
+   * @brief Retrieve the @ref SupportedMeshType, @ref type_, of this object.
+   * @return the @ref SupportedMeshType, @ref type_, of this object.
```

Comment:
Not sure if you need to repeat the `@ref SupportedMeshType` in the doc block -- the type can be clicked in the function signature as well (unless it's undocumented, which isn't).

Same elsewhere.